test/e2e/tests/import-tokens.spec.js

```diff
@@ -1,41 +1,44 @@
 const { strict: assert } = require('assert');
 const {
+  defaultGanacheOptions,
   withFixtures,
-  convertToHexValue,
-  regularDelayMs,
   unlockWallet,
 } = require('../helpers');
 const FixtureBuilder = require('../fixture-builder');
 
 describe('Import flow', function () {
-  const ganacheOptions = {
-    accounts: [
-      {
-        secretKey:
-          '0x7C9529A67102755B7E6102D6D950AC5D5863C98713805CEC576B945B15B71EAC',
-        balance: convertToHexValue(25000000000000000000),
-      },
-    ],
-  };
-
   it('allows importing multiple tokens from search', async function () {
     await withFixtures(
       {
         fixtures: new FixtureBuilder().build(),
-        ganacheOptions,
+        ganacheOptions: defaultGanacheOptions,
         title: this.test.fullTitle(),
       },
       async ({ driver }) => {
         await unlockWallet(driver);
 
         // Token list is only on mainnet
         await driver.clickElement('[data-testid="network-display"]');
+        const networkSelectionModal = await driver.findVisibleElement(
+          '.mm-modal',
+        );
+
         await driver.clickElement({ text: 'Ethereum Mainnet', tag: 'p' });
 
         // Wait for network to change and token list to load from state
-        await driver.delay(regularDelayMs);
+        await networkSelectionModal.waitForElementState('hidden');
+        if (process.env.MULTICHAIN) {
+          await driver.findVisibleElement(
+            '[data-testid="multichain-token-list-item-secondary-value"]',
```

Comment:
What is the purpose of finding the currency?

Reply:
This is to wait until network is switched. when it is still switching, we will see a loader and no secondary value. here's the temp solution to make sure we wait until network is switched then we click "import" button

Reply:
Updates: i changed the way and used `network-display` to assert so we are not differentiating MULTICHAIN with this one. [cdb6626](https://github.com/MetaMask/metamask-extension/pull/22548/commits/cdb6626a8828fee737cc23f3face0904ddc519ab)

Reply:
Hmm, I still don't follow why we need any wait step here. When the loader is on the screen, the "Import tokens" button is not clickable, so the step where that button is clicked should suffice on its own. The `clickElement` function waits until the target is both located and clickable.

Reply:
Oh right, it's because the loading overlay _isn't_ up initially, and that's still the case in this PR. Waiting for confirmation of network switch makes sense then.